### PR TITLE
Default to essential ClientInfo for JetStream

### PR DESCRIPTION
**Pull request by derekcollison**
We did default to full sharing for an export from the system account, the main one being JetStream. 

As part of this process moved server to RW lock.

Was also going to replace json encoder/decoder but I ran several experiments with my top choices and results are not what I wanted them to be. We should do custom marshallers in a follow up PR.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by matthiashanel — APPROVED** [review]
LGTM

File: server/server.go [hunk 1]
```diff
@@ -1137,13 +1137,17 @@ func (s *Server) NumLoadedAccounts() int {
 
 // LookupOrRegisterAccount will return the given account if known or create a new entry.
 func (s *Server) LookupOrRegisterAccount(name string) (account *Account, isNew bool) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	s.mu.RLock()
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/server.go:1140]: The `s.registerAccount(acc)` call below grabs the write lock, but I see your point on multiple contexts falling through, Will revert.

[the remark this answers is not in the document]
Comment (derekcollison) [reply server/server.go]: ok done.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**